File: src/Tarea7-Unidad2-Python/bank.py
class Bank:
    def __init__(self):
        self.employees = []

    def add_employee(self, employee):
        self.employees.append(employee)

    def get_all_employees(self):
        return self.employees

    def get_employee_by_name(self, name, last_name):
        for employee in self.employees:
            if employee.name.lower() == name.lower() and employee.last_name.lower() == last_name.lower():
                return employee
        return None

    def deposit_money(self, name, last_name, account_number, amount):
        employee = self.get_employee_by_name(name, last_name)
        if employee:
            for account in employee.accounts:
                if account.account_number == account_number:
                    account.deposit_money(amount)
                    return True
        return False

    def withdraw_money(self, name, last_name, account_number, amount):
        employee = self.get_employee_by_name(name, last_name)
        if employee:
            for account in employee.accounts:
                if account.account_number == account_number:
                    account.withdraw_money(amount)
                    return True
        return False

    def get_balance(self, name, last_name, account_number):
        employee = self.get_employee_by_name(name, last_name)
        if employee:
            for account in employee.accounts:
                if account.account_number == account_number:
                    return account.balance
        return -1

File: src/Tarea7-Unidad2-Java/Bank.java
import java.util.ArrayList;

public class Bank {
    private ArrayList<Employee> employees;

    public Bank() {
        this.employees = new ArrayList<>();
    }

    public void addEmployee(Employee employee) {
        employees.add(employee);
    }

    public ArrayList<Employee> getAllEmployees() {
        return employees;
    }

    public Employee getEmployeeByName(String name, String lastName) {
        for (Employee employee : employees) {
            if (employee.getName().equalsIgnoreCase(name) && employee.getLastName().equalsIgnoreCase(lastName)) {
                return employee;
            }
        }
        return null;
    }

    public boolean depositMoney(String name, String lastName, long accountNumber, double amount) {
        Employee employee = getEmployeeByName(name, lastName);
        if (employee != null) {
            ArrayList<BankAccount> accounts = employee.getAccounts();
            for (BankAccount account : accounts) {
                if (account.getAccountNumber() == accountNumber) {
                    account.depositMoney(amount);
                    return true;
                }
            }
        }
        return false;
    }

    public boolean withdrawMoney(String name, String lastName, long accountNumber, double amount) {
        Employee employee = getEmployeeByName(name, lastName);
        if (employee != null) {
            ArrayList<BankAccount> accounts = employee.getAccounts();
            for (BankAccount account : accounts) {
                if (account.getAccountNumber() == accountNumber) {
                    account.withdrawMoney(amount);
                    return true;
                }
            }
        }
        return false;
    }

    public double getBalance(String name, String lastName, long accountNumber) {
        Employee employee = getEmployeeByName(name, lastName);
        if (employee != null) {
            ArrayList<BankAccount> accounts = employee.getAccounts();
            for (BankAccount account : accounts) {
                if (account.getAccountNumber() == accountNumber) {
                    return account.getBalance();
                }
            }
        }
        return -1;
    }
}